File: lab3/lab3_Java/src/Task2.java
import java.util.Scanner;

public class Task2 {

    public static void task2() {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Введіть значення a:");
        double a = scanner.nextDouble();

        System.out.println("Введіть значення b:");
        double b = scanner.nextDouble();

        System.out.println("Введіть значення c:");
        double c = scanner.nextDouble();

        System.out.println("Введіть значення d:");
        double d = scanner.nextDouble();

        double stepX = (b - a) / 7; // Крок для x
        double stepY = (d - c) / 7; // Крок для y

        System.out.print("\t");
        double currentX = a;
        while (currentX <= b) {
            System.out.printf("%.2f\t", currentX);
            currentX += stepX;
        }
        System.out.println();

        double currentY = c;
        while (currentY <= d) {
            System.out.printf("%.2f\t", currentY);
            double innerCurrentX = a;
            while (innerCurrentX <= b) {
                double u = Math.sqrt(3 * innerCurrentX * innerCurrentX + 2 * currentY * currentY);
                System.out.printf("%.2f\t", u);
                innerCurrentX += stepX;
            }
            currentY += stepY;
            System.out.println();
        }
    }
}

File: lab3/lab3_Kotlin/src/task2.kt
import kotlin.math.sqrt
import java.util.Scanner

fun task2() {
    val scanner = Scanner(System.`in`)

    println("Введіть значення a:")
    val a = scanner.nextDouble()

    println("Введіть значення b:")
    val b = scanner.nextDouble()

    println("Введіть значення c:")
    val c = scanner.nextDouble()

    println("Введіть значення d:")
    val d = scanner.nextDouble()

    val stepX = (b - a) / 7 // Крок для x
    val stepY = (d - c) / 7 // Крок для y

    print("\t")
    var currentX = a
    while (currentX <= b) {
        print("%.2f\t".format(currentX))
        currentX += stepX
    }
    println()

    var currentY = c
    while (currentY <= d) {
        print("%.2f\t".format(currentY))
        var currentX = a
        while (currentX <= b) {
            val u = sqrt(3 * currentX * currentX + 2 * currentY * currentY)
            print("%.2f\t".format(u))
            currentX += stepX
        }
        currentY += stepY
        println()
    }
}
